pyvista/plotting/plotting.py

```diff
@@ -3499,20 +3500,58 @@ def orbit():
         return
 
 
-    def export_vtkjs(self, filename, compress_arrays=False):
+    def export_vtkjs(self, output_dir, archive=True):
         """Export the current rendering scene as a VTKjs scene.
 
-        It can be used for rendering in a web browser.
+        This constructs a scene graph from the render window and generates an
+        archive for vtk-js for rendering in a web browser.
+
+        Parameters
+        ----------
+        output_dir : str
+            The output directory for the scene graph archive.
+
+        archive : bool
+            Default True. If true, the archive will be compressed into a single
+            archive file for vtk-js and an extension of ``.vtkjs`` will be
+            added to the output directory.
 
         """
         if not hasattr(self, 'ren_win'):
             raise RuntimeError('Export must be called before showing/closing the scene.')
-        if isinstance(pyvista.FIGURE_PATH, str) and not os.path.isabs(filename):
-            filename = os.path.join(pyvista.FIGURE_PATH, filename)
+        if isinstance(pyvista.FIGURE_PATH, str) and not os.path.isabs(output_dir):
+            output_dir = os.path.join(pyvista.FIGURE_PATH, output_dir)
         else:
-            filename = os.path.abspath(os.path.expanduser(filename))
-        return export_plotter_vtkjs(self, filename, compress_arrays=compress_arrays)
-
+            output_dir = os.path.abspath(os.path.expanduser(output_dir))
+        # Check VTK version as this was introduced in VTK 9
+        if vtk.vtkVersion().GetVTKMajorVersion() < 9:
+            raise ValueError("VTK version 9 at least required.")
+        if os.path.exists(output_dir):
+            shutil.rmtree(output_dir)
```

Comment:
This could be harmful to a user's working directory if they pass in a directory where they have other data stored. The assumption here is that it must be a new directory.

Reply:
Why not write temp files in the temp directory and not remove them?  That’s much safer and matches the behavior of most programs.